### PR TITLE
fix vulnerabilities in docker image

### PR DESCRIPTION
**Pull request by ravisingal**
1. fix vulnerabilities in docker image
2. add capability to mount existing claim in pinot controller pods
3. update prometheus-jmx-exporter image version

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -13,19 +13,20 @@ RUN curl -L https://archive.apache.org/dist/pinot/apache-pinot-$PINOT_VERSION/ap
...
-    rm -rf $PINOT_HOME/plugins/*
+    rm -rf $PINOT_HOME/plugins/* && \
+    rm -rf $PINOT_HOME/plugins-external/*
 
 # Fetch jar
 RUN curl -L -o $PINOT_HOME/lib/pinot-all-${JITPACK_TAG}-shaded.jar \
         https://jitpack.io/com/github/${JITPACK_REPO}/pinot-distribution/${JITPACK_TAG}/pinot-distribution-${JITPACK_TAG}-shaded.jar
 
 # Fetch plugin jars
-RUN for artifactId in pinot-kafka-2.0 pinot-kinesis pinot-thrift pinot-json pinot-parquet pinot-orc pinot-csv pinot-confluent-avro pinot-avro pinot-protobuf pinot-batch-ingestion-standalone pinot-batch-ingestion-hadoop pinot-batch-ingestion-spark pinot-hdfs pinot-adls pinot-gcs pinot-s3 pinot-minion-builtin-tasks pinot-segment-uploader-default pinot-segment-writer-file-based pinot-dropwizard pinot-yammer; do \
```
Comment (ravisingal) [inline Dockerfile:23]: Removed the following 3 plugins having critical vulnerabilities:
1. pinot-input-format/pinot-orc
2. pinot-input-format/pinot-parquet
3. pinot-batch-ingestion/pinot-batch-ingestion-spark

File: Dockerfile [hunk 2]
```diff
@@ -13,19 +13,20 @@ RUN curl -L https://archive.apache.org/dist/pinot/apache-pinot-$PINOT_VERSION/ap
...
+RUN for artifactId in pinot-kafka-2.0 pinot-kinesis pinot-thrift pinot-json pinot-csv pinot-confluent-avro pinot-avro pinot-protobuf pinot-batch-ingestion-standalone pinot-batch-ingestion-hadoop pinot-hdfs pinot-adls pinot-gcs pinot-s3 pinot-minion-builtin-tasks pinot-segment-uploader-default pinot-segment-writer-file-based pinot-dropwizard pinot-yammer; do \
       curl -L -o $PINOT_HOME/plugins/${artifactId}-${JITPACK_TAG}-shaded.jar \
           https://jitpack.io/com/github/${JITPACK_REPO}/${artifactId}/${JITPACK_TAG}/${artifactId}-${JITPACK_TAG}-shaded.jar; \
     done
 
-FROM openjdk:11-jre-slim
```
Comment (ravisingal) [inline Dockerfile:28]: openjdk images are deprecated. https://github.com/docker-library/openjdk/issues/505

File: Dockerfile (hunk 1 above)
Comment (tim-mwangi) [reply Dockerfile:23]: Assuming we are not using them?

File: helm/templates/controller/statefulset.yaml [hunk 3]
```diff
@@ -133,9 +133,15 @@ spec:
         - name: log-config
           configMap:
             name: {{ include "pinot.controller.fullname" . }}-log-config
-        {{- if not .Values.controller.persistence.enabled }}
+        {{- if .Values.controller.persistence.existingClaim }}
```
Comment (ravisingal) [inline helm/templates/controller/statefulset.yaml:136]: if `existingClaim` is defined, it will be used instead of creating a new PVC.

**Review by tim-mwangi — APPROVED** [review]

File: helm/values.yaml [hunk 4]
```diff
@@ -457,7 +458,7 @@ server:
       port: 7071
       image:
         repository: hypertrace/prometheus-jmx-exporter
-        tag: 0.1.0
+        tag: 0.1.1
```
Comment (ravisingal) [inline helm/values.yaml:461]: https://github.com/hypertrace/prometheus-jmx-exporter/pull/1

**Review by laxmanchekka — DISMISSED** [review]

File: helm/templates/controller/statefulset.yaml [hunk 5]
```diff
@@ -45,13 +45,19 @@ spec:
               exitCode=1
               i=0
               while [ $i -le {{ .Values.zookeeper.retries }} ]; do
-                bin/zookeeper-shell.sh ZooKeeper -server {{ include "zookeeper.url" . | quote }} create {{ .Values.zookeeper.path | quote }} ""
+                bin/zookeeper-shell.sh ZooKeeper -server {{ include "zookeeper.url" . | quote }} ls {{ .Values.zookeeper.path | quote }}
```
Comment (ravisingal) [inline helm/templates/controller/statefulset.yaml:48]: new image returns error if the node already exist in zookeeper.
```
WatchedEvent state:SyncConnected type:None path:null
Node already exists: /pinot
[2022-10-20 16:45:45,596] ERROR Exiting JVM with code 1 (org.apache.zookeeper.util.ServiceUtils)
```

updated script to first check the node before creating it.

**Review by tim-mwangi — APPROVED** [review]

**Review by laxmanchekka — APPROVED** [review]

**Pull request merged**